src/john.c

```diff
@@ -858,30 +858,30 @@ static void john_load_conf(void)
 	}
 
 	if (options.activewordlistrules == NULL)
-		if (!(options.activewordlistrules =
+	  if (!(options.activewordlistrules = (char*)
 		      cfg_get_param(SECTION_OPTIONS, NULL,
 		                    "BatchModeWordlistRules")))
 			options.activewordlistrules =
 				str_alloc_copy(SUBSECTION_WORDLIST);
 
 	if (options.activesinglerules == NULL)
-		if (!(options.activesinglerules =
+		if (!(options.activesinglerules = (char*)
```

Comment:
I'm making `activesinglerules` and `activewordlistrules` `const char` in options.h and dropping these casts. This also involves adding const to `rpp_init()`